test/es6/proxy-issue884.js

```diff
@@ -0,0 +1,24 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+var proxyHandler = {
+    has() { WScript.Echo("has"); return true; }
```

Comment:
Yes, this order will be stable because it comes from the order of checks/accesses in the C++ code, not the canonical ordering of those named properties.

I'll make the change to this test.